programs/dynamic-bonding-curve/src/state/virtual_pool.rs

```diff
@@ -399,7 +399,11 @@ impl VirtualPool {
             }
         }
 
-        require!(amount_left == 0, PoolError::NotEnoughLiquidity);
+        // allow pool swallow an extra amount
+        require!(
+            amount_left <= config.get_max_swallow_quote_amount()?,
```

Comment:
I think instead of fixed 20% of the `migration_quote_threshold`, it's better to allow user to pass in the `max_allowed_amount_left: u64` as an `Option` in the `swap` endpoint. This provides more granular control for the user to configure the stop loss threshold based on their budget.

Reply:
I think minimum out amount is already a guard?

Reply:
It's different? Minimum out include slippage from market movement, while `max_allowed_amount_left` precisely control the loss.